test/testio.cpp

```diff
@@ -4142,6 +4142,20 @@ class TestIO : public TestFixture {
         ASSERT_EQUALS("[test.cpp:2]: (warning) %d in format string (no. 1) requires 'int' but the argument type is 'const void *'.\n"
                       "[test.cpp:2]: (warning) %d in format string (no. 2) requires 'int' but the argument type is 'const void *'.\n", errout.str());
 
+        check("void foo() {\n"
+              "typedef SSIZE_T ssize_t;\n" // In MSVC, SSIZE_T is available in capital letters using #include <BaseTsd.h>
```

Comment:
Good start.

You don't need the typedef. It's already defined in windows.cfg.  I'm not sure which one takes precedent. You should have tests with and without the typedef just to be safe.  There is also the problem of someone adding: `typedef SSIZE_T char;`.  Do we want to validate SSIZE_T is really ssize_t?

The problem is windows.cfg is always included and you want to test with and without it.  One way to fix that is to create two different settings (windows and not windows) in TestIO::run and select which one in TestIO::check based on platform type.  Unfortunately that doesn't let you test a windows platform without windows.cfg. This is going to be tricky to get right. This is a preexisting problem that should be worked out first.

You only added false positive tests.  You need %zu and others for false negatives.

You need to test other platforms besides windows.

You have a big matrix of configurations (with and without typedef, with and without windows.cfg, windows and non windows platforms, multiple format specifiers).

Sorry but doing this right is going to be complicated.